src/Calendar/CalendarProvider.php

```diff
@@ -0,0 +1,613 @@
+<?php
+
+namespace Sherl\Sdk\Calendar;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\RequestOptions;
+
+use Psr\Http\Message\ResponseInterface;
+
+use Sherl\Sdk\Common\Error\SherlException;
+use Sherl\Sdk\Common\Error\ErrorFactory;
+use Sherl\Sdk\Common\Error\ErrorHelper;
+use Sherl\Sdk\Calendar\Errors\CalendarErr;
+use Exception;
+use Sherl\Sdk\Common\SerializerFactory;
+
+// DTOS
+use Sherl\Sdk\Calendar\Dto\CalendarDto;
+use Sherl\Sdk\Calendar\Dto\AvailabilityDto;
+use Sherl\Sdk\Calendar\Dto\CalendarEventDto;
+use Sherl\Sdk\Calendar\Dto\CalendarFilterInputDto;
+use Sherl\Sdk\Calendar\Dto\CheckDatesInputDto;
+use Sherl\Sdk\Calendar\Dto\CheckLocationInputDto;
+use Sherl\Sdk\Calendar\Dto\CreateCalendarEventInputDto;
+use Sherl\Sdk\Calendar\Dto\CreateCalendarInputDto;
+use Sherl\Sdk\Calendar\Dto\FindAvailabilitiesInputDto;
+use Sherl\Sdk\Calendar\Dto\FindAvailabilitiesOutputDto;
+use Sherl\Sdk\Calendar\Dto\GetCalendarEventForCalendarInputDto;
+use Sherl\Sdk\Calendar\Dto\GetCalendarEventForCalendarResultsDto;
+use Sherl\Sdk\Calendar\Dto\GetCalendarEventForCurrentPersonInputDto;
+use Sherl\Sdk\Calendar\Dto\CalendarEventsPaginatedResultDto;
+use Sherl\Sdk\Calendar\Dto\UpdateCalendarEventInputDto;
+use Sherl\Sdk\Calendar\Dto\UpdateCalendarInputDto;
+use Sherl\Sdk\Calendar\Dto\GetCalendarEventByOwnerInputDto;
+
+class UserProvider
+{
+    public const DOMAIN = "Calendar";
+
+    private Client $client;
+
+    private ErrorFactory $errorFactory;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+        $this->errorFactory = new ErrorFactory(self::DOMAIN, CalendarErr::$errors);
+    }
+
+    // Calendar
+
+    /**
+     * Creates a calendar request.
+     *
+     * @param CreateCalendarInputDto $calendarData The calendar data.
+     * @throws SherlException If there is an error on creating the calendar.
+     * @return CalendarDto|null The created calendar.
+     */
+    public function createCalendardRequest(CreateCalendarInputDto $calendarData): ?CalendarDto
+    {
+        try {
+            $response = $this->client->post('/api/calendar', [
+              "headers" => [
+                "Content-Type" => "application/json",
+              ],
+              RequestOptions::JSON => [
```

Comment:
```suggestion
              RequestOptions::JSON => $calendarData
```

Est-ce qu'on pourrait passer directement l'objet ?

Reply:
J'ai des tests en PHP, ça à l'air d'être ok pour faire ça.